lib/chef/http/basic_client.rb

```diff
@@ -97,7 +97,7 @@ def request(method, url, req_body, base_headers={})
 
       #adapted from buildr/lib/buildr/core/transports.rb
       def proxy_uri
-        proxy = Chef::Config["#{url.scheme}_proxy"]
+        proxy = ENV["#{url.scheme}_proxy"] || Chef::Config["#{url.scheme}_proxy"]
```

Comment:
We should find out if the uppercase version might be a thing, maybe on windows? @lamont-granquist added uppercase env vars to our existing code that set proxy env vars in here: https://github.com/chef/chef/commit/b6b0a7ab02212a98ac6c3e100bcd9dafcf604aca but there's nothing in the commit about why uppercase was necessary.